File: sdk/php/src/Resources/VNeIdeKyc.php
<?php

namespace HiTechCloud\SDK\Resources;

class VNeIdeKyc extends BaseResource
{
    /**
     * Start personal eKYC
     */
    public function startPersonalEkyc(array $params = []): array
    {
        return $this->http->get('/api/vneidekyc/ekyc/session', $params);
    }

    /**
     * Get eKYC session
     */
    public function getEkycSession(string $sessionHash): array
    {
        return $this->http->get('/api/vneidekyc/ekyc/session/' . $sessionHash);
    }

    /**
     * List client sessions
     */
    public function listClientSessions(int $clientId): array
    {
        return $this->http->get('/api/vneidekyc/ekyc/list/' . $clientId);
    }

    /**
     * Upload file
     */
    public function uploadFile(string $sessionHash, array $data): array
    {
        return $this->http->post('/api/vneidekyc/ekyc/' . $sessionHash . '/upload', $data);
    }

    /**
     * Submit eKYC
     */
    public function submitEkyc(string $sessionHash, array $data): array
    {
        return $this->http->post('/api/vneidekyc/ekyc/' . $sessionHash . '/submit', $data);
    }

    /**
     * Cancel session
     */
    public function cancelSession(string $sessionHash): array
    {
        return $this->http->post('/api/vneidekyc/ekyc/' . $sessionHash . '/cancel');
    }

    /**
     * Accept eKYC
     */
    public function acceptEkyc(string $sessionHash): array
    {
        return $this->http->post('/api/vneidekyc/ekyc/' . $sessionHash . '/accept');
    }

    /**
     * Reject eKYC
     */
    public function rejectEkyc(string $sessionHash, array $data): array
    {
        return $this->http->post('/api/vneidekyc/ekyc/' . $sessionHash . '/reject', $data);
    }

    /**
     * Lookup organization
     */
    public function lookupOrganization(array $data): array
    {
        return $this->http->post('/api/vneidekyc/org/lookup', $data);
    }

    /**
     * Start org verification
     */
    public function startOrgVerification(array $data): array
    {
        return $this->http->post('/api/vneidekyc/org/start', $data);
    }

    /**
     * Get organization
     */
    public function getOrganization(string $orgHash): array
    {
        return $this->http->get('/api/vneidekyc/org/' . $orgHash);
    }

    /**
     * Accept organization
     */
    public function acceptOrganization(string $orgHash): array
    {
        return $this->http->post('/api/vneidekyc/org/' . $orgHash . '/accept');
    }

    /**
     * Reject organization
     */
    public function rejectOrganization(string $orgHash, array $data): array
    {
        return $this->http->post('/api/vneidekyc/org/' . $orgHash . '/reject', $data);
    }

    /**
     * Upload signed PDF
     */
    public function uploadSignedPdf(array $data): array
    {
        return $this->http->post('/api/vneidekyc/pdf/upload', $data);
    }

    /**
     * Get signed PDF
     */
    public function getSignedPdf(string $pdfHash): array
    {
        return $this->http->get('/api/vneidekyc/pdf/' . $pdfHash);
    }

    /**
     * Get client status
     */
    public function getClientStatus(int $clientId): array
    {
        return $this->http->get('/api/vneidekyc/client/' . $clientId . '/status');
    }

}

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Vneidekyc.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Vneidekyc extends BaseResource {

    public Vneidekyc(HttpClient http) {
        super(http);
    }

    /** Create eKYC session */
    public Map<String, Object> ekycSession() throws Exception {
        return http.get("/api/vneidekyc/ekyc/session");
    }

    /** Get session */
    public Map<String, Object> getSession(String sessionHash) throws Exception {
        return http.get(String.format("/api/vneidekyc/ekyc/session/%sessionHash", sessionHash));
    }

    /** List sessions */
    public Map<String, Object> listSessions(int clientId) throws Exception {
        return http.get(String.format("/api/vneidekyc/ekyc/list/%clientId", clientId));
    }

    /** Upload document */
    public Map<String, Object> upload(String sessionHash, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/vneidekyc/ekyc/%sessionHash/upload", sessionHash), data);
    }

    /** Submit */
    public Map<String, Object> submit(String sessionHash) throws Exception {
        return http.post(String.format("/api/vneidekyc/ekyc/%sessionHash/submit", sessionHash), null);
    }

    /** Cancel */
    public Map<String, Object> cancel(String sessionHash) throws Exception {
        return http.post(String.format("/api/vneidekyc/ekyc/%sessionHash/cancel", sessionHash), null);
    }

    /** Accept */
    public Map<String, Object> accept(String sessionHash) throws Exception {
        return http.post(String.format("/api/vneidekyc/ekyc/%sessionHash/accept", sessionHash), null);
    }

    /** Reject */
    public Map<String, Object> reject(String sessionHash) throws Exception {
        return http.post(String.format("/api/vneidekyc/ekyc/%sessionHash/reject", sessionHash), null);
    }

    /** Org lookup */
    public Map<String, Object> orgLookup(Map<String, Object> data) throws Exception {
        return http.post("/api/vneidekyc/org/lookup", data);
    }

    /** Start org verification */
    public Map<String, Object> orgStart(Map<String, Object> data) throws Exception {
        return http.post("/api/vneidekyc/org/start", data);
    }

    /** Get org session */
    public Map<String, Object> orgSession(String orgHash) throws Exception {
        return http.get(String.format("/api/vneidekyc/org/%orgHash", orgHash));
    }

    /** Accept org */
    public Map<String, Object> orgAccept(String orgHash) throws Exception {
        return http.post(String.format("/api/vneidekyc/org/%orgHash/accept", orgHash), null);
    }

    /** Reject org */
    public Map<String, Object> orgReject(String orgHash) throws Exception {
        return http.post(String.format("/api/vneidekyc/org/%orgHash/reject", orgHash), null);
    }

    /** Upload PDF */
    public Map<String, Object> pdfUpload(Map<String, Object> data) throws Exception {
        return http.post("/api/vneidekyc/pdf/upload", data);
    }

    /** Get PDF status */
    public Map<String, Object> pdfStatus(String pdfHash) throws Exception {
        return http.get(String.format("/api/vneidekyc/pdf/%pdfHash", pdfHash));
    }

    /** Get client status */
    public Map<String, Object> clientStatus(int clientId) throws Exception {
        return http.get(String.format("/api/vneidekyc/client/%clientId/status", clientId));
    }

}
